app/scripts/metamask-controller.js

```diff
@@ -3129,7 +3258,10 @@ export default class MetamaskController extends EventEmitter {
         this.preferencesController.getLedgerTransportPreference();
       this.setLedgerTransportPreference(transportPreference);
 
-      this.selectFirstIdentity();
+      this.selectFirstAccount();
+      this.accountsController.setSelectedAccount(
```

Comment:
Is this, and the `firstAccount` variable, needed since we just called `selectFirstAccount`?

Reply:
No, we can actually remove that. Removed on [141b1ef](https://github.com/MetaMask/metamask-extension/pull/21437/commits/141b1efe6e71e12d03a064d8c2588680a7846dc3)